influxql/iterator.gen.go.tmpl

```diff
@@ -370,6 +371,84 @@ func (itr *{{.name}}LimitIterator) Next() *{{.Name}}Point {
 	}
 }
 
+// {{.name}}FillIterator represents a fill iterator that processes {{.name}} values.
+type {{.name}}FillIterator struct {
+	name  string
+	input {{.Name}}Iterator
+	buf   *{{.Name}}Point
+	prev  *{{.Name}}Point
+	next  int64
+	opt   IteratorOptions
+	fill  func (*{{.name}}FillIterator) {{.Type}}
+}
+
+func new{{.Name}}FillIterator(input {{.Name}}Iterator, opt IteratorOptions) *{{.name}}FillIterator {
+	measurement := opt.Sources[0].(*Measurement)
+	var fill func (*{{.name}}FillIterator) {{.Type}}
+	switch opt.Fill {
+	case NullFill:
+		fill = func (*{{.name}}FillIterator) {{.Type}} {
+			return {{.Nil}}
+		}
+	case NumberFill:
+		v, ok := opt.FillValue.({{.Type}})
+		if !ok {
+			panic("invalid fill value for type {{.Type}}")
+		}
+		fill = func (*{{.name}}FillIterator) {{.Type}} {
+			return v
+		}
+	case PreviousFill:
+		fill = func (itr *{{.name}}FillIterator) {{.Type}} {
+			if itr.prev != nil {
+				return itr.prev.Value
+			}
+			return {{.Nil}}
+		}
+	default:
+		panic(fmt.Sprintf("unknown fill option: %T", opt.Fill))
+	}
+
+	return &{{.name}}FillIterator{
+		name:  measurement.Name,
+		input: input,
+		next:  opt.StartTime + int64(opt.Interval.Offset),
+		opt:   opt,
+		fill:  fill,
+	}
+}
+
+func (itr *{{.name}}FillIterator) Close() error { return itr.input.Close() }
+
+func (itr *{{.name}}FillIterator) Next() *{{.Name}}Point {
+	pt := itr.buf
```

Comment:
Minor nit: can you use `p` instead of `pt` for consistency with the rest of the code?